python/foglamp/services/core/api/plugins/common.py

```diff
@@ -119,39 +120,84 @@ def load_and_fetch_c_hybrid_plugin_info(plugin_name: str, is_config: bool, plugi
+    # Require a local import in order to avoid circular import references
+    from foglamp.services.core import server
+
+    stdout_file_path = create_log_file(action="list")
+    tmp_log_output_fp = stdout_file_path.split('logs/')[:1][0] + "logs/output.txt"
+    _platform = platform.platform()
+    pkg_type = "" if package_type is None else package_type
+    pkg_mgt = 'apt'
+    ret_code = 0
+    category = await server.Server._configuration_manager.get_category_all_items("Installation")
+    max_update_cat_item = category['maxUpdate']
+    pkg_cache_mgr = server.Server._package_cache_manager
+    last_accessed_time = pkg_cache_mgr['update']['last_accessed_time']
+    now = datetime.now()
+    then = last_accessed_time if last_accessed_time else now
+    duration_in_sec = (now - then).total_seconds()
+    # If max update per day is set to 1, then an update can not occurs until 24 hours after the last accessed update.
+    # If set to 2 then this drops to 12 hours between updates, 3 would result in 8 hours between calls and so on.
+    if duration_in_sec > (24 / int(max_update_cat_item['value'])) * 60 * 60 or not last_accessed_time:
+        _logger.info("Attempting update on {}".format(now))
+        cmd = "sudo {} -y update > {} 2>&1".format(pkg_mgt, stdout_file_path)
+        if 'centos' in _platform or 'redhat' in _platform:
+            pkg_mgt = 'yum'
+            cmd = "sudo {} check-update > {} 2>&1".format(pkg_mgt, stdout_file_path)
+
+        ret_code = os.system(cmd)
+        if ret_code == 0:
+            pkg_cache_mgr['update']['last_accessed_time'] = now
+            # fetch available package caching always clear on every update request
+            _get_available_packages.cache_clear()
+    else:
+        _logger.warning("Maximum update exceeds the limit for the day")
+
+    ttl_cat_item_val = int(category['listAvailablePackagesCacheTTL']['value'])
+    if ttl_cat_item_val > 0:
+        last_accessed_time = pkg_cache_mgr['list']['last_accessed_time']
+        now = datetime.now()
+        if not last_accessed_time:
+            last_accessed_time = now
+            pkg_cache_mgr['list']['last_accessed_time'] = now
+        duration_in_sec = (now - last_accessed_time).total_seconds()
+        if duration_in_sec > ttl_cat_item_val * 60:
+            _get_available_packages.cache_clear()
+            pkg_cache_mgr['list']['last_accessed_time'] = datetime.now()
+    else:
+        _get_available_packages.cache_clear()
+        pkg_cache_mgr['list']['last_accessed_time'] = ""
+
+    ret_code, available_packages = _get_available_packages(ret_code, tmp_log_output_fp, pkg_mgt, pkg_type)
 
     # combine above output in logs file
     with open("{}".format(stdout_file_path), 'a') as fh:
-        fh.write(" \n".join(log_output))
+        fh.write(" \n".join(available_packages))
+        if not len(available_packages):
+            fh.write("No package available to install")
```

Comment:
removed check and appending to available_packages; otherwise  _get_available_packages shall result API to return  {"plugins": ["No package available to install"], "link": "log/191025-16-47-43-list.log"}